tests/integ/scala/test_snowflake_plan_suite.py

```diff
@@ -208,15 +219,20 @@ def test_create_scoped_temp_table(session):
             == f' CREATE  TEMPORARY  TABLE {temp_table_name}("NUM" BIGINT, "STR" STRING(8))'
         )
         assert (
-            session._plan_builder.create_temp_table(
-                temp_table_name,
+            session._plan_builder.save_as_table(
+                [temp_table_name],
+                None,
+                SaveMode.APPEND,
+                "temp",
+                None,
+                None,
                 df._plan,
                 use_scoped_temp_objects=True,
                 is_generated=False,
             )
             .queries[0]
             .sql
-            == f' CREATE  TEMPORARY  TABLE {temp_table_name}("NUM" BIGINT, "STR" STRING(8))'
+            == f' CREATE  TEMPORARY  TABLE {temp_table_name} If  NOT  EXISTS ("NUM" BIGINT, "STR" STRING(8))'
```

Comment:
this is not a breaking change because 
1. in case of `cache_result`, we never called with with `is_generated=False`
2. when `is_generated=False` i.e. table creation is explicitely requested by users, then for save mode append, we add `if not exists` in the old behavior.